posts/2023-10-31-23.0.0.11-beta.adoc

```diff
@@ -0,0 +1,330 @@
+* link:/docs/latest/reference/feature/mpMetrics-5.0.html[MicroProfile Metrics 5.0 feature]
+* link:/docs/latest/metrics-list.html[Metrics reference list]
+    
+
+// DO NOT MODIFY THIS LINE. </GHA-BLOG-TOPIC> 
+
+// // // // DO NOT MODIFY THIS COMMENT BLOCK <GHA-BLOG-TOPIC> // // // // 
+// Blog issue: https://github.com/OpenLiberty/open-liberty/issues/26642
+// Contact/Reviewer: abutch3r
+// // // // // // // // 
+
+[#mpreact]
+== Negative acknowledgement and more for MicroProfile Reactive Messaging 3.0 and MicroProfile Stream Operators 3.0
+
+MicroProfile Reactive Messaging 3.0 has a number of new features and changes over MicroProfile Reactive Messaging 1.0, including negative acknowledgements, emitters, and backpressure support.
+
+An application using MicroProfile Reactive Messaging or MicroProfile Stream Operators is typically composed of CDI beans that consume, produce, and process messages that are passing along reactive streams. These messages can be internal to the application, or they can be sent and received through external message brokers. MicroProfile Reactive Messaging uses MicroProfile Stream Operators to pass messages through channels between methods or to messaging solutions such as Kafka to provide resilient storage of messages.
+
+MicroProfile Reactive Messaging and Reactive Stream Operators 3.0 provide a range of new features and is compatible with Jakarta EE 9 onwards.
+
+To use both features together, add the Microprfile Reactive Messaging 3.0 feature to the `server.xml`:
+
+[source,xml]
+----
+<feature>mpReactiveMessaging-3.0</feature>
+----
+
+Enabling MicroProfile Reactive Messaging 3.0 automatically enables the MicroProfile Reactive Stream Operators 3.0 feature too.
+
+To use MicroProfile Reactive Stream Operators 3.0 without MicroProfile Reactive Messaging:
+
+[source,xml]
+----
+<feature>mpReactiveStreams-3.0</feature>
+----
+
+
+=== Negative acknowledgements
+
+In MicroProfile Reactive Messaging 1.0, Messages could only be positively acknowledged. If there was a problem with the payload or if exceptional behavior occurred, there was no mechanism to indicate the problem or to handle the problem if it occurred within the stream. The new _nack_ support can send or handle these events.
+
+To explicitly negatively acknowledge an incoming Message:
+
+[source,java]
+----
+@Incoming("data")
+@Outgoing("out")
+public Message<String> process(Message<String> m) { 
+    String s = m.getPayload();
+    if (s.equalsIgnoreCase("b")) {
+        // we cannot fail, we must nack explicitly.
+        m.nack(new IllegalArgumentException("b"));
+        return null;
+    }
+    return m.withPayload(s.toUpperCase());
+}
+----
+
+The method signature of accepting a Message indicates that the acknowledgement strategy for this method is `MANUAL`. It is the responsibility of your code to `ack()` or `nack()` the message. In the above example, the message can be acknowledged downstream of the out channel.
```

Comment:
```suggestion
The method signature of accepting a Message indicates that the acknowledgement strategy for this method is `MANUAL`. It is the responsibility of your code to `ack()` or `nack()` the message. In the previous example, the message can be acknowledged downstream of the out channel.
```

Reply:
> The method signature of accepting a Message indicates that the acknowledgement strategy for this method is `MANUAL`

how do we know this? is it in the code sample or offered as a clarification in the description?